packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -22,90 +27,25 @@ import { unlock } from '../lock-unlock';
  * @return {WPHigherOrderComponent} Higher-order component.
  */
 
-const BLOCK_BINDINGS_ALLOWED_BLOCKS = {
-	'core/paragraph': [ 'content' ],
-	'core/heading': [ 'content' ],
-	'core/image': [ 'id', 'url', 'title', 'alt' ],
-	'core/button': [ 'url', 'text', 'linkTarget', 'rel' ],
-};
-
-/**
- * Based on the given block name,
- * check if it is possible to bind the block.
- *
- * @param {string} blockName - The block name.
- * @return {boolean} Whether it is possible to bind the block to sources.
- */
-export function canBindBlock( blockName ) {
-	return blockName in BLOCK_BINDINGS_ALLOWED_BLOCKS;
-}
-
-/**
- * Based on the given block name and attribute name,
- * check if it is possible to bind the block attribute.
- *
- * @param {string} blockName     - The block name.
- * @param {string} attributeName - The attribute name.
- * @return {boolean} Whether it is possible to bind the block attribute.
- */
-export function canBindAttribute( blockName, attributeName ) {
-	return (
-		canBindBlock( blockName ) &&
-		BLOCK_BINDINGS_ALLOWED_BLOCKS[ blockName ].includes( attributeName )
-	);
-}
-
 export const withBlockBindingSupport = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
 		const registry = useRegistry();
-		const sources = useSelect( ( select ) =>
-			unlock( select( blocksStore ) ).getAllBlockBindingsSources()
-		);
-		const bindings = props.attributes.metadata?.bindings;
-		const { name, clientId, context } = props;
-		const boundAttributes = useSelect( () => {
-			if ( ! bindings ) {
-				return;
-			}
-
-			const attributes = {};
-
-			for ( const [ attributeName, boundAttribute ] of Object.entries(
-				bindings
-			) ) {
-				const source = sources[ boundAttribute.source ];
-				if (
-					! source?.getValue ||
-					! canBindAttribute( name, attributeName )
-				) {
-					continue;
-				}
-
-				const args = {
-					registry,
-					context,
-					clientId,
-					attributeName,
-					args: boundAttribute.args,
-				};
+		const {
+			attributes: blockAttributes,
+			name,
+			clientId,
+			context,
+			setAttributes,
+		} = props;
+		const bindings = blockAttributes?.metadata?.bindings;
 
-				attributes[ attributeName ] = source.getValue( args );
-
-				if ( attributes[ attributeName ] === undefined ) {
-					if ( attributeName === 'url' ) {
-						attributes[ attributeName ] = null;
-					} else {
-						attributes[ attributeName ] =
-							source.getPlaceholder?.( args );
-					}
-				}
-			}
-
-			return attributes;
-		}, [ bindings, name, clientId, context, registry, sources ] );
-
-		const { setAttributes } = props;
+		const boundAttributes = useSelect( () => {
+			return getBoundAttributesValues( clientId, context, registry );
```

Comment:
Why is this a reusable function if it's only used once? I also don't really like this, it becomes unclear what store this is selecting from.

Reply:
It was meant to be used in other places like [allow editing of post meta](https://github.com/WordPress/gutenberg/pull/61753), but we changed the approach there so it is not reused anymore. See [related comment](https://github.com/WordPress/gutenberg/pull/62063#issuecomment-2138907668).